### PR TITLE
Disable change heap config option

### PR DESCRIPTION
**Pull request by Shiney**
This is a fix that creates the config flag for #65 it also removes some assertions that were firing in #1062 as I could not get the tests to work correctly with those assertions failing.

### CODE REVIEW

File: lib/Runtime/Base/FunctionBody.cpp [hunk 1]
```diff
@@ -1520,7 +1520,7 @@ namespace Js
 
     ScriptFunctionType * FunctionProxy::EnsureDeferredPrototypeType()
     {
-        Assert(this->GetFunctionProxy() == this);
+        //Assert(this->GetFunctionProxy() == this);
```
Comment (MikeHolman) [inline lib/Runtime/Base/FunctionBody.cpp]: we need to fix the assert, we can't just disable it

Comment (Shiney) [inline lib/Runtime/Base/FunctionBody.cpp]: Okay.

I haven't really been able to work out what that assert is for, and it seems to be possible to make it happen without the other changes in this hotfix. (See #1062)

**Comment by EdMaurer** [comment]
@Shiney should we close this out or do you want to keep going with it?

**Comment by Shiney** [comment]
Sorry I can't work on this my current contract doesn't allow me to
contribute intellectual property except for my employer.

**Pull request closed without merging**